### PR TITLE
honggfuzz: Use LLVM 12

### LINKED ISSUE #174118: honggfuzz: hfuzz-clang generates broken executables
### Describe the bug

Binaries built with `hfuzz-clang` fail during startup with `ERROR: UndefinedBehaviorSanitizer failed to allocate 0x0 (0) bytes of SetAlternateSignalStack`. `hfuzz-gcc` works correctly.
Seems to be related to: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100114

### Steps To Reproduce
1. `nix shell nixpkgs#honggfuzz`
2. `echo "int main() { return 0; }" > test.c`
3.  `hfuzz-clang test.c -o test`
4. `./test`

```
==2252345==ERROR: UndefinedBehaviorSanitizer failed to allocate 0x0 (0) bytes of SetAlternateSignalStack (error code: 22)
==2252345==Process memory map follows:
[...]
==2252345==End of process memory map.
==2252345==Sanitizer CHECK failed: /build/compiler-rt-11.1.0.src/lib/sanitizer_common/sanitizer_common.cpp:54 ((0 && "unable to mmap")) != (0) (0, 0)
```
### Expected behavior
Compiled binary should execute normally.

### Notify maintainers

@cpu

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.7, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by chivay**
###### Description of changes
Using LLVM 11 results in broken binaries generated by hfuzz-clang,
crashing with error:
UndefinedBehaviorSanitizer failed to allocate 0x0 (0) bytes of SetAlternateSignalStack

Fixes #174118

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chivay** [comment]
CC @cpu

**Review by cpu — APPROVED** [review]
Thank you @chivay - I haven't been doing any fuzzing work for some time now and so this package has suffered some bitrot :-) Are you interested in becoming a co-maintainer (or taking the maintainer slot entirely)?

This fix looks good to me in theory but I won't have a chance to test it locally in practice for a little while.

**Comment by chivay** [comment]
> Thank you @chivay - I haven't been doing any fuzzing work for some time now and so this package has suffered some bitrot :-) Are you interested in becoming a co-maintainer (or taking the maintainer slot entirely)?

Sure, I'll add myself as the maintainer.

**Review by cpu — APPROVED** [review]

File: pkgs/tools/security/honggfuzz/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/security/honggfuzz/default.nix:51]: Why do we have this let in here?

Comment (chivay) [reply pkgs/tools/security/honggfuzz/default.nix:51]: Seems unnecessary to me. Shall it remove it now or create a separate PR?

Comment (cpu) [reply pkgs/tools/security/honggfuzz/default.nix:51]: Agreed - I think it was just confusion on my part in the original commit.

I vote remove it as part of this PR but defer to anyone with merge privs if they feel otherwise :-P

Comment (SuperSandro2000) [reply pkgs/tools/security/honggfuzz/default.nix:51]: I'll just do it after the merge.

**Pull request merged**
